src/server/main_service.cc

```diff
@@ -859,8 +860,13 @@ void Service::DispatchCommand(CmdArgList args, facade::ConnectionContext* cntx)
 
       if (!dist_trans->IsMulti()) {  // Multi command initialize themself based on their mode.
         if (auto st = dist_trans->InitByArgs(dfly_cntx->conn_state.db_index, args_no_cmd);
-            st != OpStatus::OK)
+            st != OpStatus::OK) {
+          if (cmd_name == "ZUNION" && st == OpStatus::INVALID_INT) {
```

Comment:
If we don't catch it here explicitly for `zunion`, the error message shown on terminal would be `syntax error` which is not exactly same as redis message.
`zunionstore` also follows the same error message convention.

Reply:
why the error message text is important for you @rounaknandanwar ?

Reply:
I would assume consistency with redis. It's not a must have but it's a good thing to be consistent -- makes sense from a UX perspective.

Reply:
@rounaknandanwar  says that `zunionstore` follows the same convention and yet we do not handle this case here.
Also, I prefer avoid spreading the error handling logic in many places and here it's a new special case that is introduced. Instead I suggest returning a new OpStatus enum and handle it inside   `SendError`

Reply:
@rounaknandanwar yep, I am in favor of not splitting the error logic as well because it fragments the error handling. My comment was more of what their rational was. But I agree with you, @rounaknandanwar could you plz take care of this.

Reply:
1. Yes the intent of the specific message was consistency with redis. Is it a hard requirement to keep the message same ? I can see for other commands as well, we don't exactly replicate the messaging. Example for the command `zunionstore out -1 key`, the error messages are different:
     - Dragonfly : `ERR value is not an integer or out of range`
     - Redis : `ERR at least 1 input key is needed for zunionstore`
     
2. I kind of agree that the error logic should not be fragmented so I was thinking if each command should have its own error handling. But I guess some of the validations are done early because of transaction specific logic so that cannot be achieved completely as @kostasrim pointed out.

3. zunionstore returns the message - `(error) ERR syntax error`. I think I misread it earlier. Should we keep it uniform for zunion as well then ?

4. I discovered another version of zunionstore that crashes - `zunionstore out 0  aggregate`. I can handle this in this PR itself.

Reply:
Just remove the `if (cmd_name == "ZUNION" && st == OpStatus::INVALID_INT) {`. 

You have basically two choices: 
1. Either add a new error code as @romange suggested in `OpStatus` enum.
2. Leave it as is, which I personally don't mind -- unless @romange has a strong objection (we can't always be 1-1 with redis errors). 

> I discovered another version of zunionstore that crashes - zunionstore out 0  aggregate. I can handle this in this PR itself.

Definitely, it would be a nice additions ;)

Reply:
Keeping it uniform and showing message as `(error) ERR syntax error` for both `zunion` and `zunionstore`.